src/apps/weblib/interface.cpp

```diff
@@ -373,15 +369,24 @@ void Interface::update(ObjectRegistry::Handle chart,
 	::Anim::TimePoint time(nanoSecs);
 
 	widget->getChart().getAnimControl().update(time);
+}
 
-	if (render) {
-		const Geom::Size size{width, height};
-		auto ptr = objects.get<Vizzu::Main::JScriptCanvas>(canvas);
-		ptr->frameBegin();
-		widget->onUpdateSize(size);
-		widget->onDraw(ptr);
-		ptr->frameEnd();
-	}
+void Interface::render(ObjectRegistry::Handle chart,
+    ObjectRegistry::Handle canvas,
+    double width,
+    double height)
+{
+	auto &&widget = objects.get<UI::ChartWidget>(chart);
+	auto ptr = objects.get<Vizzu::Main::JScriptCanvas>(canvas);
+
+	ptr->frameBegin();
+
+	const Geom::Size size{width, height};
+	widget->onUpdateSize(size);
```

Comment:
inline `size` variable, or make the c++ Interface parameter as `Geom::Size` replacement of 2 doubles (width, height).